Library/Homebrew/test/utils/linkage_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "utils/linkage"
+
+RSpec.describe Utils do
+  [:needs_macos, :needs_linux].each do |needs_os|
+    describe "::binary_linked_to_library?", needs_os do
```

Comment:
Given it's now a unit test: this shouldn't need a particular OS any more and would be good to fix so it also works in generic mode

Reply:
`binary_linked_to_library?` calls `dynamically_linked_libraries` which is not defined generically but rather has OS-specific implementations. We could consider providing a generic `dynamically_linked_libraries`, but I'm not sure what it should do (what type of executable object should be analyzed in a generic context?).

Reply:
Ah, all good then, sorry!